x/tokenomics/keeper/relay_mining_difficulty.go

```diff
@@ -0,0 +1,71 @@
+package keeper
+
+import (
+	"context"
+
+	"cosmossdk.io/store/prefix"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+
+	"github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+// SetRelayMiningDifficulty set a specific relayMiningDifficulty in the store from its index
+func (k Keeper) SetRelayMiningDifficulty(ctx context.Context, relayMiningDifficulty types.RelayMiningDifficulty) {
+	storeAdapter := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
+	store := prefix.NewStore(storeAdapter, types.KeyPrefix(types.RelayMiningDifficultyKeyPrefix))
+	difficultyBz := k.cdc.MustMarshal(&relayMiningDifficulty)
+	store.Set(types.RelayMiningDifficultyKey(
+		relayMiningDifficulty.ServiceId,
+	), difficultyBz)
+}
+
+// GetRelayMiningDifficulty returns a relayMiningDifficulty from its index
+func (k Keeper) GetRelayMiningDifficulty(
+	ctx context.Context,
+	serviceId string,
+
+) (difficulty types.RelayMiningDifficulty, found bool) {
+	storeAdapter := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
+	store := prefix.NewStore(storeAdapter, types.KeyPrefix(types.RelayMiningDifficultyKeyPrefix))
+
+	difficultyBz := store.Get(types.RelayMiningDifficultyKey(
+		serviceId,
+	))
+	if difficultyBz == nil {
+		return difficulty, false
```

Comment:
```suggestion
		difficulty.ServiceId = serviceId
		return difficulty, false
```

This should return a difficulty with the right `serviceId`